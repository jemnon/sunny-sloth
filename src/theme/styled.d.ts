import 'styled-components';
import {
  Breakpoints,
  Colors,
  ColorMode,
  FocusColors,
  Fonts,
  FontSizes,
  Header,
  HoverColors,
  MaxWidths,
  Spacing,
  ZIndex,
} from '../types/theme';

declare module 'styled-components' {
  export interface DefaultTheme {
    breakpoints: Breakpoints;
    colors: Colors;
    focusColors: FocusColors;
    fonts: Fonts;
    fontSizes: FontSizes;
    header: Header;
    hoverColors: HoverColors;
    maxWidths: MaxWidths;
    spacing: Spacing;
    transition: string;
    zIndex: ZIndex;
  }
}
