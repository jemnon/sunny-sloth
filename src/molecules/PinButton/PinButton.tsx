import React, { FC } from 'react';
import styled from 'styled-components';
import Button from '../../atoms/Button';
import { Pinterest } from '../../atoms/Icons';

interface PinButtonProps {}

const PinButtonContainer = styled.div`
  display: inline-flex;
  justify-content: center;
  align-items: center;
`;

const PinButtonLabel = styled.span`
  padding-right: ${({ theme }): string => theme.spacing.sm2};
`;

const PinButton: FC<PinButtonProps> = () => (
  <PinButtonContainer>
    <Button variant="outline" shape="rectangle" size="small">
      <PinButtonLabel>Pin It</PinButtonLabel>
      <Pinterest
        fontSize="0.85rem"
        fill="#BB5B34"
        style={{ marginTop: '2px' }}
      />
    </Button>
  </PinButtonContainer>
);

export default PinButton;
