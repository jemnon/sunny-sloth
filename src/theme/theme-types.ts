export interface Breakpoints {
  sm: string;
  md: string;
  lg: string;
  xlg: string;
}

export interface Fonts {
  lato: string;
  latoBold: string;
  latoItalic: string;
  noto: string;
  notoBold: string;
}

export interface Colors {
  cream: string;
  darkTeal: string;
  teal: string;
  lightBrown: string;
  nearBlack: string;
  nearWhite: string;
  darkOrange: string;
  orange: string;
  white: string;
  red: string;
  green: string;
}

export type ColorMode = 'dark' | 'light';

export interface MaxWidths {
  sm: string;
  md: string;
  lg: string;
}

export interface Spacing {
  sm1: string;
  sm2: string;
  sm3: string;
  sm4: string;
  md1: string;
  md2: string;
  md3: string;
  md4: string;
  lg1: string;
  lg2: string;
  lg3: string;
  lg4: string;
}

export interface ZIndex {
  z0: string;
  z1: string;
  z2: string;
  z3: string;
  z4: string;
  z5: string;
  z6: string;
  z999: string;
  z9999: string;
}
