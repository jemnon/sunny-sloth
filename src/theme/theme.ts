const theme = {
  breakpoints: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
    xlg: '1440px',
  },
  fonts: {
    latoBold:
      `"Lato-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato-Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    notoBold:
      `"Noto Serif TC-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
  },
  spacing: {
    sm1: '0.25rem' /* 4px */,
    sm2: '0.5rem' /*  8px */,
    sm3: '0.75rem' /* 12px */,
    sm4: '1rem' /* 16px */,
    md1: '1.25rem' /* 20px */,
    md2: '1.5rem' /* 24px */,
    md3: '1.75rem' /* 28px */,
    md4: '2rem' /* 32px */,
    lg1: '2.25rem' /* 36px */,
    lg2: '2.5rem' /* 40px */,
    lg3: '2.75rem' /* 44px */,
    lg4: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
