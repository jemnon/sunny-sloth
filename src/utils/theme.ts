import { DefaultTheme } from 'styled-components';

const theme: DefaultTheme = {
  breakpoints: {
    /* TODO: remove once conversion to styled-breakpoints is done */
    desktop: 'screen and (min-width: 960px)',
    tablet: 'screen and (min-width: 480px) and (max-width: 959px)',
    mobile: 'screen and (max-width: 479px)',
    /* ^^ */
    sm: '480px',
    md: '960px',
    lg: '1440px',
    xlg: '1920px',
  },
  fonts: {
    latoBold:
      `"Lato Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1440px',
  },
  spacing: {
    s1: '0.25rem' /* 4px */,
    s2: '0.5rem' /*  8px */,
    s3: '0.75rem' /* 12px */,
    s4: '1rem' /* 16px */,
    s5: '1.25rem' /* 20px */,
    s6: '1.5rem' /* 24px */,
    s7: '1.75rem' /* 28px */,
    s8: '2rem' /* 32px */,
    s9: '2.25rem' /* 36px */,
    s10: '2.5rem' /* 40px */,
    s11: '2.75rem' /* 44px */,
    s12: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
