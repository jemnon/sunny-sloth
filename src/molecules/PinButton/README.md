# PinButton

### Description
Component description

```js
import PinButton from '@fenderdigital/ui/FILL IN TYPEPinButton';
```

### Props
* `propName` - property name and description 
