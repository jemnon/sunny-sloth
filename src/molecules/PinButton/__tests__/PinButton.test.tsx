import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import PinButton from '../PinButton';
import theme from '../../../theme';

describe('<PinButton />', (): void => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <PinButton />
    </ThemeProvider>
  );
  it('exists', (): void => {
    expect(PinButton).toBeDefined();
  });
  it('renders defaults correctly', (): void => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});