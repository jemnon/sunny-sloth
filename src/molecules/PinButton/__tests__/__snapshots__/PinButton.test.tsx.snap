// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PinButton /> renders defaults correctly 1`] = `
<div
  className="sc-dJjZJu gJuiWY"
>
  <button
    className="sc-bdvvaa fVbsZy"
    onClick={[Function]}
    shape="rectangle"
    size="small"
  >
    <span
      className="sc-hGPAah bvQlhi"
    >
      Pin It
    </span>
    <svg
      className="sc-pVTma eKehlp"
      fill="#BB5B34"
      fontSize="0.85rem"
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "marginTop": "2px",
        }
      }
      viewBox="0 0 17 17"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 8.5a8.5 8.5 0 005.4 7.91 8.34 8.34 0 010-2.44c.16-.66 1-4.22 1-4.22a3.11 3.11 0 01-.25-1.26c0-1.18.68-2.06 1.53-2.06a1.06 1.06 0 011.11 1.19 17 17 0 01-.7 2.83A1.23 1.23 0 009.34 12c1.51 0 2.67-1.6 2.67-3.9a3.35 3.35 0 00-3.54-3.46 3.68 3.68 0 00-3.84 3.69 3.27 3.27 0 00.63 1.93.27.27 0 01.06.25c-.07.27-.21.85-.24 1s-.12.19-.28.12A3.82 3.82 0 013.07 8.3c0-2.68 1.93-5.14 5.62-5.14a5 5 0 015.24 4.9c0 2.93-1.85 5.29-4.41 5.29a2.29 2.29 0 01-1.95-1S7.14 14 7 14.39a9.2 9.2 0 01-1 2.23 8.4 8.4 0 002.5.38A8.5 8.5 0 100 8.5z"
      />
    </svg>
  </button>
</div>
`;
